RELEASE.md

```diff
@@ -1,66 +1,78 @@
-## Github Binary Releases
+We consider `dev` to be the branch where the source code of `HEAD` always reflects a state with the latest delivered development changes for the next release.
 
-# Versioning
+When the source code in the `dev` branch reaches a stable point and is ready to be released, all the changes are merged back into `main` and then tagged with a release number while bumping `MAJOR` and/or `MINOR`.
+
+## Feature Branches
+
+New features are developed into separate branches that only live in the contributor's forks.
+
+- branch off from: `dev`
+- merge back into: `dev`
+- naming convention: `feat-x`, where `x` describes the feature
+
+## Patch Branches
+
+Bugs are patched into separate branches that only live in the contributor's forks.
+
+- branch off from: `main`
+- merge back into: `main` + `dev`
+- naming convention: `patch-x`, where `x` describes the bug/patch
+
+# Releasing Roles Binaries
+
+The [release page of SRI repo](https://github.com/stratum-mining/stratum/releases) provides executable binaries for all SRI roles, targeting popular hardware architectures.
+
+The github binary releases of the roles are handled in the `release-bin.yaml` workflow.
+
+This workflow is manually started by navigating to the "Actions" tab in the SRI repo, then navigating to the Release workflow section, and clicking "Run Workflow".
+
+Note: in order to be able to manually trigger the "Run Workflow" button, the user needs to have "Write" permissions on the repository, otherwise the button will not show up on the UI.
+
+# Publishing Library Crates
+
+Although SRI has a global release cycle, which is attached to the binaries, each internal crate also has its own versioning history.
+
+Lib crates are published to crates.io in the `release-lib.yaml` workflow. The workflow tries to update all the library crates. 
+If a crate is not to updated, the step will fail for that each step have continue-on-error set to true.
+
+Since each step can fail, the output ot the action must be manually check to macke sure that all the library intended to
+be published are published.
 
-Versioning is handled by the `autorelease.yaml` workflow. The workflow will  will auto detect changes in crates and auto bump the patch versions 
-in the target crate, as well as any crate that uses the target crate as a dependency, if changes are detected. It will then push a new commit with the 
-versioning changes to the `bot/versioning` branch, and Lastly, it will auto create a PR (bot/versioning -> main). Since cargo-smart-release does not yet 
-support autodetection of MAJOR and MINOR changes with regards to Semver, any MAJOR/MINOR versioning will need to be manually changed in the package Cargo.toml.
+Running `cargo release` in the various workspace help to prepare the version number and everything.
 
-## Versioning Notes
+Every PR needs to increase the version of whatever crate it is touching. Otherwise, we will mess up the dependency chain of whoever is fetching from crates.io
```

Comment:
I would specify, every PR **in main**. PRs that add feature in dev are not going to increase the version number since they will be batched every [duration] into main. So lets say that when I do the release I add 5 feature to lib1 I will not increase the minor version of 5 but of 1.

So when we merge dev in main we increase the versions of the touched libs. When we merge a buge fix in main we increase the version of the touched libs. Whenever we merge something in main we run the releases actions